### PR TITLE
Added SageMaker batch transform support.

### PR DESCRIPTION
**Pull request by jaheba**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

TODO:

- [x] add tests
- [x] add checks 
    * ensure that the right parameters are set? (is this needed)
- [ ] add graceful error handling

---

To use batch inference, one has to set the environment variable: `INFERENCE_CONFIG` to the right json string.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-317/1 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-317/1/index.html

**Comment by szha** [comment]
Job PR-317/2 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-317/2/index.html

**Comment by szha** [comment]
Job PR-317/3 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-317/3/index.html

File: src/gluonts/model/forecast.py [hunk 1]
```diff
@@ -556,9 +556,12 @@ class OutputType(str, Enum):
 
 
 class Config(pydantic.BaseModel):
-    num_eval_samples: int = pydantic.Schema(..., alias="num_samples")
-    output_types: Set[OutputType]
-    quantiles: List[str]  # FIXME: validate list elements
+    num_eval_samples: int = pydantic.Schema(100, alias="num_samples")
+    output_types: Set[OutputType] = {"qunatiles", "mean"}
```
Comment (dcmaddix) [inline src/gluonts/model/forecast.py]: Fix spelling of `qunatiles` to `quantiles`.

Comment (jaheba) [reply src/gluonts/model/forecast.py]: Oh, thanks!

**Comment by szha** [comment]
Job PR-317/4 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-317/4/index.html

File: src/gluonts/shell/serve/__init__.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (vafl) [inline src/gluonts/shell/serve/__init__.py]: Can we maybe put this in another file and just import the stuff we want to expose here?
This has been our convention so far regarding `__init__.py`.

File: src/gluonts/shell/serve/util.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import numpy as np
+
+
+def jsonify_floats(json_object):
```
Comment (vafl) [inline src/gluonts/shell/serve/util.py]: We could subclass `json.JSONEncoder` -- but this is fine too.

File: src/gluonts/shell/serve/__init__.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    def stop(self, *args, **kwargs):
+        logger.info("Shutting down GluonTS scoring service")
+
+
+def make_gunicorn_app(
+    env: ServeEnv,
+    forecaster_type: Optional[Type[Union[Estimator, Predictor]]],
+    settings: Settings,
+) -> Application:
+    check_gpu_support()
```
Comment (vafl) [inline src/gluonts/shell/serve/__init__.py]: I guess for supporting gpu's properly, we need a bit more logic for handling the number of proceses.

Also, the server should probably be started with the following environment variables to ensure that each server has one cpu and mxnet does not try to parallelize a single request. 
```
OMP_NUM_THREADS=1
MXNET_ENGINE_TYPE=NaiveEngine
```

Comment (jaheba) [reply src/gluonts/shell/serve/__init__.py]: Thanks. These changes were just moved around, but we should still do what you’ve suggested.

**Comment by szha** [comment]
Job PR-317/5 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-317/5/index.html

**Pull request merged**